app/src/main/java/com/example/bigowlapp/activity/SupervisedGroupListActivity.java

```diff
@@ -46,14 +48,21 @@ private void subscribeToData() {
                                     supervisedGroupsListView = findViewById(R.id.supervised_groups);
                                     supervisedGroupsListView.setAdapter(new SupervisedGroupAdaptor(getBaseContext(), new ArrayList<>(supervisedGroups)));
                                     supervisedGroupsListView.setOnItemClickListener((arg0, v, position, arg3) -> {
-                                        Intent intent = new Intent(getBaseContext(), SupervisedGroupPageActivity.class);
+                                        intent = new Intent(getBaseContext(), SupervisedGroupPageActivity.class);
                                         intent.putExtra("groupID", supervisedGroups.get(position).getUid());
                                         intent.putExtra("groupName", supervisedGroups.get(position).getName());
                                         intent.putExtra("supervisorName", supervisorNameList.get(position));
                                         startActivity(intent);
                                     });
-                                } else
-                                    this.noGroupAlert().show();
+                                } else {
+                                    this.noGroupAlert = new AlertDialog.Builder(SupervisedGroupListActivity.this)
+                                            .setTitle("No supervised group found!")
+                                            .setMessage("Required to be a supervised user of any group")
+                                            .setPositiveButton("Ok", (dialogInterface, which) -> SupervisedGroupListActivity.super.onBackPressed())
```

Comment:
not an issue, but more for my curiosity. is there a reason that super `onBackPressed()` is used instead of just the class itself `this.onBackPressed()`?